Test_Engine/Query/FullMessage.cs

```diff
@@ -0,0 +1,101 @@
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Base;
+using BH.oM.Test;
+using BH.oM.Test.Results;
+
+namespace BH.Engine.Test
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns a full concatenated message for the test results and its inner result to the target max depth, only giving messages worse or equal to the provided severity level.")]
+        [Input("result", "The TestResult to give the full message for.")]
+        [Input("maxDepth", "Maximum level of recursiveness for inner TestInformation.")]
+        [Input("minSeverity", "The minimum level of severity of the information to report back.")]
+        [Output("message", "Full message for the TestResult")]
+        public static string FullMessage(this TestResult result, int maxDepth = 3, TestStatus minSeverity = TestStatus.Warning)
+        {
+            //Return an empty string if the status is not at least as severe as the provided minimum.
+            if (!result.Status.IsEqualOrMoreSevere(minSeverity))
+                return "";
+
+            string message = result.Description + Environment.NewLine;
+            message += result.Message + Environment.NewLine;
+
+            if (maxDepth > 0 && result.Information.Count > 0)
+            {
+                message += "Inner results:" + Environment.NewLine;
```

Comment:
Thanks! Indenting is actually already happening. The reason it can not be seen in GH comes down to the panels in GH I think. IF you try copying over the text from the panel to for example notepad, it should be indented!

Changing the text the "Results at depth x" is tricky with the current implementation of the recursive setup. Could ofc be refactored to allow for it, but kind of like the current flow..

Would you mind checking if the indentation in something other than a GH Panel looks ok for you?

Reply:
OK indenting looks good in sublime - an we get a line space!
![Screenshot 2021-01-26 140128](https://user-images.githubusercontent.com/6618854/105854719-24856600-5fdf-11eb-8844-006bf1c1ba86.jpg)
I can see the difficulty in passing around the current depth - would be nice but not essential.

Reply:
Also I note the indenting also indicates the depth of the Inner results so I am happy to discard the proposal of enumerating depth.
![Screenshot 2021-01-26 140735](https://user-images.githubusercontent.com/6618854/105855453-010eeb00-5fe0-11eb-9f27-47bf2b9f5295.jpg)